File: 15. Apriori.R
## Apriori ##


# libraries ---------------------------------------------------------------

library(tidyverse)
library(arules)
library(arulesViz)


# read data ---------------------------------------------------------------

data <- arules::read.transactions("data/Market_Basket_Optimisation.csv", 
                                  sep = ",", rm.duplicates = TRUE)

summary(data)

arules::itemFrequencyPlot(data, topN = 10) # 10 productos más comprados

# para hacer el gráfico bonito, puedo acceder a los datos así:
frec <- arules::itemFrequency(data) %>% 
  as.data.frame() %>% 
  tibble::rownames_to_column()

# entrenar algoritmo ------------------------------------------------------

rules <- arules::apriori(data, 
                         parameter = list(support = 0.004, confidence = 0.2))

# Reglas fuertes
# Visualización de resultados

arules::inspect(arules::sort(rules, by = "lift")[1:10])

plot(rules, method = "graph", engine = "htmlwidget")

File: 15. Apriori.py
# -*- coding: utf-8 -*-
"""
Created on Mon Oct 11 14:58:05 2021

@author: Admin
"""

## Apriori ##

# =============================================================================
# libraries
# =============================================================================

import pandas as pd

# =============================================================================
# read data
# =============================================================================

data = pd.read_csv("data\Market_Basket_Optimisation.csv", header=None)

transactions = []

for i in range(0, 7501): 
    transactions.append([str(data.values[i, j]) for j in range(0, 20)])
    
# =============================================================================
# entrenar algoritmo
# =============================================================================

from apyori import apriori
rules = apriori(transactions, min_support = 0.003, min_confidence = 0.2,
                min_length = 2, min_lift = 3)


# =============================================================================
# Visualización resultados 
# =============================================================================

results = list(rules)

results[0]
